helix-term/src/commands/typed.rs

```diff
@@ -1365,21 +1363,63 @@ fn lsp_stop(
 
     let doc = doc!(cx.editor);
 
-    let ls_id = doc
-        .language_server()
-        .map(|ls| ls.id())
-        .context("LSP not running for the current document")?;
+    // TODO this stops language servers which may be used in another doc/language type that uses the same language servers
+    // I'm not sure if this is really what we want
+    let ls_shutdown_names = doc
+        .language_servers()
+        .iter()
+        .map(|ls| ls.name())
```

Comment:
Well it is, but I'm not really sure if it really improves the situation, as the stop function would be significantly bigger (and requires an extra clone for the ls name, because of ownership issues):
```rust
    pub fn stop_by_id(&mut self, id: usize) {
        let ls_name = self
            .inner
            .values()
            .find(|client| client.id() == id)
            .map(|client| (*client.name()).to_string());

        if let Some(name) = ls_name {
            if let Some(client) = self.inner.remove(&name) {
                tokio::spawn(async move { client.force_shutdown().await.ok() });
            }
        }
    }
```

Reply:
I see, yeah, that all makes sense; thanks for explaining, guys 👍